tools/src/test/java/org/apache/kafka/tools/TopicCommandIntegrationTest.java

```diff
@@ -86,652 +90,792 @@
+            topicService.deleteTopic(deleteOpts);
 
-        if (!isKRaftTest()) {
-            String deletePath = kafka.zk.DeleteTopicsTopicZNode.path(testTopicName);
-            assertFalse(zkClient().pathExists(deletePath), "Delete path for topic shouldn't exist before deletion.");
+            TestUtils.waitForCondition(
+                    () -> adminClient.listTopics().listings().get().stream().noneMatch(topic -> topic.name().equals(testTopicName)),
+                    60000, "Delete topic fail in 60000 ms"
+            );
         }
-        topicService.deleteTopic(deleteOpts);
-        TestUtils.verifyTopicDeletion(zkClientOrNull(), testTopicName, 1, brokers());
     }
 
-    @ParameterizedTest
-    @ValueSource(strings = {"zk", "kraft"})
-    public void testTopicWithCollidingCharDeletionAndCreateAgain(String quorum) throws Exception {
-        // create the topic with colliding chars
-        String topicWithCollidingChar = "test.a";
-        TestUtils.createTopicWithAdmin(adminClient, topicWithCollidingChar, scalaBrokers, scalaControllers, defaultNumPartitions, defaultReplicationFactor,
-                scala.collection.immutable.Map$.MODULE$.empty(), new Properties()
-        );
-        // delete the topic
-        TopicCommand.TopicCommandOptions deleteOpts = buildTopicCommandOptionsWithBootstrap("--delete", "--topic", topicWithCollidingChar);
-
-        if (!isKRaftTest()) {
-            String deletePath = kafka.zk.DeleteTopicsTopicZNode.path(topicWithCollidingChar);
-            assertFalse(zkClient().pathExists(deletePath), "Delete path for topic shouldn't exist before deletion.");
+    @ClusterTemplate("generate1")
+    public void testTopicWithCollidingCharDeletionAndCreateAgain(TestInfo testInfo) throws Exception {
+        try (Admin adminClient = clusterInstance.createAdminClient();
+             TopicCommand.TopicService topicService = new TopicCommand.TopicService(adminClient)) {
+            // create the topic with colliding chars
+            String topicWithCollidingChar = "test.a";
+            adminClient.createTopics(Collections.singletonList(new NewTopic(topicWithCollidingChar, defaultNumPartitions, defaultReplicationFactor)));
+            clusterInstance.waitForTopic(topicWithCollidingChar, defaultNumPartitions);
+
+            // delete the topic
+            TopicCommand.TopicCommandOptions deleteOpts = buildTopicCommandOptionsWithBootstrap("--delete", "--topic", topicWithCollidingChar);
+            topicService.deleteTopic(deleteOpts);
+            TestUtils.waitForCondition(
+                    () -> adminClient.listTopics().listings().get().stream().noneMatch(topic -> topic.name().equals(topicWithCollidingChar)),
+                    60000, "Delete topic fail in 60000 ms"
+            );
+
+            kafka.utils.TestUtils.verifyTopicDeletion(null, topicWithCollidingChar, 1, JavaConverters.asScalaIteratorConverter(
```

Comment:
Could you please file a MINOR patch to add `verifyTopicDeletion` to `ClusterInstance`?

Reply:
Ok. I am working on this.

Reply:
I opened a PR [here]( https://github.com/apache/kafka/pull/16627/commits/3883c4b2115be2c59c86e51fc8b67648321aa4df). Please take a look.